jaeger-core/src/main/java/com/uber/jaeger/Tracer.java

```diff
@@ -163,19 +172,17 @@ public void close() {
 
     private String operationName = null;
     private long startTimeMicroseconds;
-    private SpanContext parent;
+    private List<Reference> references = new ArrayList<Reference>();
     private final Map<String, Object> tags = new HashMap<String, Object>();
+    private final Map<String, String> baggage = new HashMap<String, String>();
```

Comment:
this introduces a couple of inefficiencies, we're always allocating two collections even though the list of references could be empty (see my comment below about child-of vs. parent_id representation), and when there is only one parent its baggage can be used as is, without copying (baggage is supposed to be immutable).